lib/cadet_web/admin_controllers/admin_user_controller.ex

```diff
@@ -289,4 +289,109 @@ defmodule CadetWeb.AdminUserController do
         end
     }
   end
+
+  def combined_user_total_xp(conn, _) do
+    course_id = conn.assigns.course_reg.course_id
+
+    combined_user_xp_total_query = """
```

Comment:
could you place this query under the `Achievements` context instead of the controller layer? i.e. `lib/cadet/incentives/achievements.ex`